### PR TITLE
Reload teleport service config on upgrade

### LINKED ISSUE #41099: Reload systemd config after an upgrade
### Expected behavior:
The systemd configuration for the teleport service should be reloaded after an upgrade.

### Current behavior:
Changes to teleport service config are not loaded after an upgrade. For example: https://github.com/gravitational/teleport/issues/40186 patch does not get reloaded after an upgrade.

### Bug details:
- Teleport version: 15.2.1
- Recreation steps:
Install Teleport < 15.2.1, then upgrade teleport to 15.2.1+. The `/lib/systemd/system/teleport.service` file will show that `RestartSec=5`, but the teleport service will still be restarting at 1sec intervals.
- Debug logs
```console
$ cat /lib/systemd/system/teleport.service
[Unit]
Description=Teleport Service
After=network.target

[Service]
RestartSec=5
...

$ teleport version
Teleport Enterprise v15.2.1 git:v15.2.1-0-g6e72fc7 go1.21.9

$ systemctl status teleport
Warning: The unit file, source configuration file or drop-ins of teleport.service changed on disk. Run 'systemctl daemon-reload' to reload units.
× teleport.service - Teleport Service
     Loaded: loaded (/lib/systemd/system/teleport.service; enabled; vendor preset: enabled)
    Drop-In: /etc/systemd/system/teleport.service.d
             └─teleport-upgrader-env.conf
     Active: failed (Result: exit-code) since Wed 2024-05-01 16:35:30 UTC; 2s ago
    Process: 39213 ExecStart=/usr/local/bin/teleport start --config /etc/teleport.yaml --pid-file=/run/teleport.pid (code=exited, status=1/FAILURE)
   Main PID: 39213 (code=exited, status=1/FAILURE)
        CPU: 152ms

May 01 16:35:30 teleport0430 systemd[1]: teleport.service: Scheduled restart job, restart counter is at 5.
May 01 16:35:30 teleport0430 systemd[1]: Stopped Teleport Service.
May 01 16:35:30 teleport0430 systemd[1]: teleport.service: Start request repeated too quickly.
May 01 16:35:30 teleport0430 systemd[1]: teleport.service: Failed with result 'exit-code'.
May 01 16:35:30 teleport0430 systemd[1]: Failed to start Teleport Service.
```

### PR DESCRIPTION
**Pull request by bernardjkim**
Closes https://github.com/gravitational/teleport/issues/41099.

This PR adds a post install script to the teleport package. This post install script runs `systemctl daemon-reload` to reload any changes that have been made to the `teleport.service`.

changelog: teleport service config is now reloaded on upgrades

### CODE REVIEW

File: examples/systemd/post-upgrade [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+# this post upgrade script is run each time the teleport package is upgraded
+
+set -eu
+
+# skip reload and restart when systemd is disabled. This is only relevant when
+# testing in a container.
+if systemctl status > /dev/null 2>&1; then
+    # reload systemd configuration
+    systemctl daemon-reload
```
Comment (codingllama) [inline examples/systemd/post-upgrade:11]: Is there a way to reload just Teleport-related configs? Does that make sense?

File: build.assets/build-package.sh [hunk 2]
```diff
@@ -290,6 +290,10 @@ if [[ "${PACKAGE_TYPE}" != "pkg" ]]; then
         mv -v ${LINUX_CONFIG_FILE} ${PACKAGE_TEMPDIR}/buildroot${LINUX_CONFIG_DIR}
         CONFIG_FILE_STANZA="--config-files /src/buildroot${LINUX_CONFIG_DIR}/${LINUX_CONFIG_FILE} "
     fi
+
+    # include post-upgrade script
+    mv -v ${TAR_PATH}/examples/systemd/post-upgrade ${PACKAGE_TEMPDIR}
```
Comment (codingllama) [inline build.assets/build-package.sh:295]: I wish we moved production-worthy code outside of examples.

No action expected here, just chatting.

**Review by codingllama — APPROVED** [review]
I'm not super familiar with this but looks sensible to me.

**Review by EdwardDowling — APPROVED** [review]

File: examples/systemd/post-upgrade (hunk 1 above)
Comment (bernardjkim) [reply examples/systemd/post-upgrade]: I'm not sure there is a way to reload just the teleport related configs. It seems like system daemon-reload is the common pattern.

File: build.assets/build-package.sh (hunk 2 above)
Comment (bernardjkim) [reply build.assets/build-package.sh:295]: Yeah, I'm a little reluctant to add this to the examples dir, but it seems like this is where the teleport systemd configuration lives.

**Pull request merged**